ddprof-lib/src/main/cpp/objectSampler.cpp

```diff
@@ -188,14 +189,15 @@ Error ObjectSampler::updateConfiguration(u64 events, double time_coefficient) {
       CONFIG_UPDATE_CHECK_PERIOD_SECS, 15);
 
   float signal = pid_controller.compute(events, time_coefficient);
-  int required_interval = _interval - static_cast<int>(signal);
+  u64 required_interval = static_cast<u64>(_interval) - static_cast<int>(signal);
```

Comment:
This is probably not OK if `interval < signal`.
You will have both sides converted to a common type, then the subtraction will wrap around (UB).

Reply:
Ok. I just clamped the `required_interval` in case `_interval` is smaller than `signal` to avoid wrap-around.